CONTRIBUTING.md

```diff
@@ -1 +1,118 @@
+
+   If your change is for fixing an issue, please write ["Fixes <issue-URL>"](https://help.github.com/articles/closing-issues-using-keywords/) in the description section of your pull request.  Github would close the issue when the owners merge your pull request.
+
+   Please remember to specify some reviewers for your pull request.  If you don't know who are the right ones, please follow Github's recommendation.
+
+
+1. Delete local and remote branches
+
+   To keep your local workspace and your fork clean, you might want to remove merged branches:
+
+   ```bash
+   git push origin :my-cool-stuff
+   git checkout develop
+   git pull upstream develop
+   git branch -d my-cool-stuff
+   ```
+
+### Good Manner in Code Review
+
+-  Please feel free to ping your reviewers by sending them the URL of your pull request via IM or email.  Please do this after your pull request passes the CI.
+
+- Please answer reviewers' every comment.  If you are to follow the comment, please write "Done"; please give a reason otherwise.
+
+- If you don't want your reviewers to get overwhelmed by email notifications, you might reply their comments by [in a batch](https://help.github.com/articles/reviewing-proposed-changes-in-a-pull-request/).
+
+- Reduce the unnecessary commits.  Some developers commit often.  It is recommended to append a sequence of small changes into one commit by running `git commit --amend` instead of `git commit`.
+
+## Logging in the Code
+
+We use [glog](https://github.com/google/glog) for logging in our C/C++ code.
+
+For general information, please use `LOG`.  For debug information, please use [`VLOG`](http://htmlpreview.github.io/?https://github.com/google/glog/blob/master/doc/glog.html#verbose).  The reason is at [here](https://groups.google.com/a/chromium.org/d/msg/chromium-dev/3NDNd1KzXeY/AZKMMx37fdQJ).
+
+`VLOG` requires a *verbose level* parameter.  For example:
+
+```c++
+VLOG(3) << "Operator FC is taking " << num_inputs << "inputs."
+```
+
+When we run a PaddlePaddle application or test, we can specify a verbose threshold.  For example:
+
+```bash
+GLOG_v=3 python ../python/paddle/v2/framework/tests/test_recurrent_op.py
+```
+
+This will enable all VLOG messages from verbose level 0 to 3, so you will see above example VLOG message, which is in level 3.  This suggests that we output overall messages in lower verbose levels, so they display with higher probability.  When coding C++, please follow the verbose level convention as follows:
+
+
+| level |                               |
+|-------|-------------------------------|
+| 0     | load shared libraries and other inter-system operations |
+| 1     | reserved                      |
+| 2     | DL model compilation, including the building of backward pass and adding optimizer, feed, init operators |
+| 3     | DL model execution.  For example, in operators |
+| 4     | More detailed information than above. |
```

Comment:
Maybe the simplest way to define the max level of the vlog is corresponding with our namespace.

`framework` should use `VLOG(1)` as its highest VLOG level. Since the framework is our core code. `VLOG(2)` and `VLOG(3)` could be used in `framework` for unimportant logs.

`operators` should use `VLOG(3)` as its highest VLOG level, since it is the middle level of our framework. `operators` will invoke platform, memories, eigen, etc.

`platform`, `memory` should use `VLOG(5)` as their highest VLOG level, since they are usually invoked at the end our code.

| namespace |  max vlog level |
| --- | ---|
| framework | 1|
| operators | 3 |
| memory/platform | 5|

Reply:
Done